nat46/modules/nat46-netdev.c

```diff
@@ -251,6 +251,47 @@ int nat46_configure(char *devname, char *buf) {
 	}
 }
 
+int nat46_remove(char *devname, char *buf) {
+	int ret = -1;
+	int buflen = 1024;
+	char config_remove[buflen];
+	struct net_device *dev;
+	nat46_instance_t *nat46;
+	nat46_instance_t *nat46_remove;
+	int result_rem;
+	int i;
+
+	if((dev = find_dev(devname)) == NULL ||
+	   (nat46 = netdev_nat46_instance(dev)) == NULL ||
+	   (nat46_remove = alloc_nat46_instance(1, NULL, -1, -1, -1)) == NULL) {
+		return ret;
+	}
+
+	if(nat46_set_ipair_config(nat46_remove, 0, buf, buflen) < 0) {
+		release_nat46_instance(nat46_remove);
+		return ret;
+	}
+
+	result_rem = nat46_get_ipair_config(nat46_remove, 0, config_remove, buflen);
+	for(i = 0; i < nat46->npairs; i++) {
+		char config[buflen];
+		int result = nat46_get_ipair_config(nat46, i, config, buflen);
+
+		if (result_rem == result && strncmp(config_remove, config, result_rem) == 0) {
+			nat46_instance_t *nat46_new = alloc_nat46_instance(nat46->npairs-1, nat46, 0, 0, i);
+			if(nat46_new) {
+				netdev_nat46_set_instance(dev, nat46_new);
```

Comment:
(2) if for some reason, there are two lines with identical config, this line will be run  twice. The first call free the nat46 structure and set a new one in its place.

Reply:
Hi,  the for loop will exit on first match or no match.  There's a break statement inside the if condition.  So once a match is found, regardless of whether or not the removal was successful, the code will break out of the loop. If there are duplicate entries, the code will only remove the first match.  Thanks.